stream.go

```diff
@@ -0,0 +1,150 @@
+package tarantool
+
+import (
+	"fmt"
+	"time"
+
+	"gopkg.in/vmihailenco/msgpack.v2"
+)
+
+type TxnIsolationLevel uint64
+
+const (
+	// By default, the isolation level of Tarantool is serializable.
+	defaultIsolationLevel = iota
+	// The ReadCommittedLevel isolation level makes visible all transactions
+	// that started commit (stream.Do(NewCommitRequest()) was called).
+	ReadCommittedLevel
+	// The ReadConfirmedLevel isolation level makes visible all transactions
+	// that finished the commit (stream.Do(NewCommitRequest()) was returned).
+	ReadConfirmedLevel
+	// If the BestEffortLevel (serializable) isolation level becomes unreachable,
+	// the transaction is marked as «conflicted» and can no longer be committed.
+	BestEffortLevel
+)
+
+type Stream struct {
+	Id   uint64
+	Conn *Connection
+}
+
+func fillBegin(enc *msgpack.Encoder, txnIsolation TxnIsolationLevel, timeout time.Duration) error {
+	hasTimeout := timeout > 0
+	hasIsolationLevel := txnIsolation != defaultIsolationLevel
+	mapLen := 0
+	if hasTimeout {
+		mapLen += 1
+	}
+	if hasIsolationLevel {
+		mapLen += 1
+	}
+
+	err := enc.EncodeMapLen(mapLen)
+
+	if hasTimeout {
+		enc.EncodeUint64(KeyTimeout)
+		err = enc.Encode(timeout.Seconds())
+	}
+
+	if hasIsolationLevel {
+		enc.EncodeUint64(KeyTxnIsolation)
+		err = enc.Encode(txnIsolation)
+	}
+
+	return err
```

Comment:
If `enc.EncodeMapLen(mapLen)` or `enc.Encode(timeout.Seconds())` had error, but `enc.Encode(txnIsolation)` don't, we will miss the error and treat this case is ok (if we ever check Encode errors, I remember ignoring at least some of them in lint PR).

Reply:
Fixed, thanks!